### ISSUE TITLE
load_locales! causes `may not live long enough` error

### ISSUE THREAD

**Issue by SergeFan** [Baptistemontan/leptos_i18n#177]
I have a Tauri App project with Leptos as frontend. I follow the book and add the macro to the `main.rs` but get the 'may not live long enough' error.

```
mod app;

use leptos::prelude::mount_to_body;
use leptos::*;
use thaw::{ConfigProvider, ToasterProvider};

use app::*;

leptos_i18n::load_locales!();
use i18n::*;

fn main() {
    console_error_panic_hook::set_once();
    mount_to_body(|| {
        view! {
            <I18nContextProvider>
                <ConfigProvider>
                    <ToasterProvider>
                        <App/>
                    </ToasterProvider>
                </ConfigProvider>
            </I18nContextProvider>
        }
    })
}
```

The full error logs are:

```
error[E0310]: the parameter type `__into_view_comp_b__` may not live long enough
 --> src/main.rs:9:1
  |
9 | leptos_i18n::load_locales!();
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  | |
  | the parameter type `__into_view_comp_b__` must be valid for the static lifetime...
  | ...so that the type `__into_view_comp_b__` will meet its required lifetime bounds
  |
  = note: this error originates in the macro `$crate::__private::macros_reexport::load_locales` which comes from the expansion of the macro `leptos_i18n::load_locales` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider adding an explicit lifetime bound
 --> /home/...../.cargo/registry/src/index.crates.io-6f17d22bba15001f/leptos_i18n-0.5.2/src/macros.rs:10:67
  |
10|         $crate::__private::macros_reexport::load_locales!{$($tt)*}: 'static
```

Am I doing wrong?

**Comment by Baptistemontan** [comment]
Don't worry you are doing everything fine, the problem comes from the macro itself. What I'm going to need from you to fix this is your translations files and your Cargo.toml if this is ok for you to share them.

**Comment by SergeFan** [comment]
The `cargo.toml` goes

```
[dependencies]
chrono = "0.4"
console_error_panic_hook = "0.1"
icondata = { version = "0.5", default-features = false, features = ["css-gg", "github-octicons"] }
js-sys = "0.3"
leptos = { version = "0.7", features = ["csr"] }
leptos_i18n = "0.5"
serde = { version = "1.0", features = ["derive"] }
serde-wasm-bindgen = "0.6"
thaw = { version = "0.4", features = ["csr"] }
wasm-bindgen = "0.2"
wasm-bindgen-futures = "0.4"

[package.metadata.leptos-i18n]
default = "en"
locales = ["en", "jp", "sc"]

[workspace]
members = ["src-tauri"]
```

The translation files don't have to much lines since I'm just trying to implement it.

`locales/en.json`

```
{
  "usage_guide": "<p>\n<b>\"Click the button\"</b>\" to select your input file\"<br/>\n\"OR\"<br/>\n<b>\"Drag & drop\"</b>\" your file to the App window\"\n</p>"
}
```

`locales/jp.json`

```
{
  "usage_guide": "<p>\n<b>\"ボタンをクリック\"</b>\"して入力ファイルを選んでください\"<br/>\n\"または\"<br/>\n\"アプリウインドウに\"<b>\"入力ファイルをドラッグ&ドロップ\"</b>\"してください\"\n</p>"
}
```

`locales/sc.json`

```
{
  "usage_guide": "<p>\n<b>\"点击按钮\"</b>\"选择输入文件路径\"<br/>\n\"或\"<br/>\n<b>\"直接拖拽文件\"</b>\"至应用窗口\"\n</p>"
}
```

**Comment by Baptistemontan** [comment]
Thanks a lot! I'll take a look into it when I get back home from holidays!

**Comment by Baptistemontan** [comment]
Just released the fix with v0.5.3
